archunit/src/main/java/com/tngtech/archunit/library/plantuml/rules/PlantUmlPatterns.java

```diff
@@ -46,7 +48,7 @@ class PlantUmlPatterns {
     private static final String COLOR_FORMAT = "\\s*(?:#" + anyOf("\\w|/\\\\-") + "+)?";
 
     private static final Pattern PLANTUML_COMPONENT_PATTERN = Pattern.compile(
-            "^\\s*" + COMPONENT_NAME_FORMAT + "\\s*" + STEREOTYPE_FORMAT + "*" + ALIAS_FORMAT + COLOR_FORMAT + "\\s*");
+            "^\\s*" + COMPONENT_TYPE_FORMAT + "\\s*"  + COMPONENT_NAME_FORMAT + "\\s*" + STEREOTYPE_FORMAT + "*" + ALIAS_FORMAT + COLOR_FORMAT + "\\s*");
```

Comment:
I think this should (at least) be a separate commit. Because it has nothing to do with the PR title "ignore unknown components in dependencies", right?

I guess you added this, because it seems inconsistent once you add `database ...` declarations to declare components just as `[foo] ...`?

I'm principally fine with supporting the `component` keyword, but then I think we should support it fully? Because if you use the keyword, you can also omit the square brackets, right? I.e. `component foo <..bar..>`. So, we would have to support both cases, with and without square brackets. (That's exactly one reason why the original supported syntax was so limited though 😉 To have a clearly defined and easy to comprehend subset without too many extra cases...)

Reply:
I guess once we open that box we also need to support dependencies that are defined from those components declared without square brackets...

```
component foo <<..any..>>
[bar] <<..other..>>

foo --> [bar]
```

Not sure, if that's the case already? 🤔  I guess it would already be valid PlantUML to define

```
[foo] <<..any..>>
[bar] <<..other..>>

foo --> bar
```

🙄

Reply:
separated commit https://github.com/TNG/ArchUnit/pull/1223/commits/39190ba798e524f8bd57472e717a0f0653a3c4f3
outside of this PR (https://github.com/TNG/ArchUnit/pull/1223)